src/main/java/org/folio/service/inventory/InventoryItemManager.java

```diff
@@ -142,18 +143,22 @@ public Future<Void> updateItemWithPieceFields(Piece piece, RequestContext reques
     if (piece.getItemId() == null || piece.getPoLineId() == null || piece.getIsBound()) {
       return Future.succeededFuture();
     }
+
     String itemId = piece.getItemId();
-    String poLineId = piece.getPoLineId();
-    return getItemRecordById(itemId, true, requestContext)
-      .compose(item -> {
-        if (poLineId == null || item == null || item.isEmpty()) {
-          return Future.succeededFuture();
-        }
-        InventoryUtils.updateItemWithPieceFields(item, piece);
-        return updateItem(item, requestContext);
-      });
+
+    return chainCall(Collections.singletonList(itemId), id ->
+      getItemRecordById(id, true, requestContext)
+        .compose(item -> {
```

Comment:
Can you please explain difference, in previous implementation we used compose and in function chainCall() we also use compose between 2 futures.